lib/parser.js

```diff
@@ -27,68 +27,52 @@ THE SOFTWARE
 var di = require('di');
 
 module.exports = ParserFactory;
+
 di.annotate(ParserFactory, new di.Provide('DHCP.parser'));
 di.annotate(ParserFactory, new di.Inject('DHCP.protocol', 'Logger', 'Assert'));
-function ParserFactory(protocol, Logger, assert) {
-    var logger = Logger.initialize(ParserFactory);
-    var parser = {};
 
-    parser.parse = function(msg) {
-        var len;
-        //logger.info(rinfo.address + ':' + rinfo.port + '/' + msg.length + 'b');
-        var packet = {
-            op: protocol.BOOTPMessageType.get(msg.readUInt8(0)),
-            // htype is combined into chaddr field object
-            hlen: msg.readUInt8(2),
-            hops: msg.readUInt8(3),
-            xid: msg.readUInt32BE(4),
-            secs: msg.readUInt16BE(8),
-            flags: msg.readUInt16BE(10),
-            ciaddr: readIpRaw(msg, 12),
-            yiaddr: readIpRaw(msg, 16),
-            siaddr: readIpRaw(msg, 20),
-            giaddr: readIpRaw(msg, 24),
-            chaddr: protocol.createHardwareAddress(
-                protocol.ARPHardwareType.get(msg.readUInt8(1)),
-                readAddressRaw(msg, 28, msg.readUInt8(2))),
-            sname: trimNulls(msg.toString('ascii', 44, 108)),
-            fname: trimNulls(msg.toString('ascii', 108, 236)),
-            magic: msg.readUInt32BE(236),
-            options: {}
-        };
-        var offset = 240;
-        var code = 0;
+function ParserFactory(protocol, Logger, assert) {
+    return {
+        logger: Logger.initialize(ParserFactory),
+        len: null,
```

Comment:
The reason `len` was moved into the parser object is because it needed to be refactored in order to expose certain privileged methods that were defined inside the `parse` method and were hoisted into the parser object in order to expose them. So they can be more easily tested.